### PR TITLE
Recover password

### PR DESCRIPTION
**Pull request by donvico**

### CODE REVIEW

File: middleware/password.recovery.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+
+const nodemailer = require('nodemailer')
+const crypto = require('crypto')
+const dotenv = require('promise-dotenv').config()
```
Comment (bolowys33) [inline middleware/password.recovery.js:4]: We are to use the normal dotenv

File: middleware/password.recovery.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+
+async function forgetPassword(req,res){
+    const {email} = req.body
+    const user = await user.findOne({email})
+    if(!user){
+        return res.status(404).json({success: false, message: 'User not found' })
+    }
+    const token = await generatePasswordResetToken()
+  user.passwordResetToken = token;
```
Comment (bolowys33) [inline middleware/password.recovery.js:19]: I don't understand this part, why are you saving the token in the user model?

File: middleware/password.recovery.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    },
+  });
+  const mailOption = {
+    from: 'cijeoma559@gmail.com',
+    to: email,
+    subject: 'Reset your password',
+    text: `Click the following link to reset your password: ${app-link}/reset-password/${token}`,
+  };
+
+  try{
```
Comment (bolowys33) [inline middleware/password.recovery.js:35]: The trycatch may  not be necessary as the sendMail method takes a callback

`transporter.sendMail(mailOptions, (error, info) => {
    if (error) {
      console.error("Error sending email: " + error);
    } else {
      console.log("Email sent: " + info.response);
    }
  });`

File: middleware/password.recovery.js [hunk 4]
```diff
@@ -1,4 +1,57 @@
 const express = require('express');
 const app = express();
 const nodemailer = require('nodemailer');
+const dotenv = require("promise-dotenv").config()
```
Comment (promise-J) [inline middleware/password.recovery.js]: you can change this to dotenv instead of promise-dotenv. 
We agreed on using the dotenv

**Review by promise-J — CHANGES_REQUESTED** [review]

**Pull request merged**